### PR TITLE
Initial commit for tasks 1-2

### PR DESCRIPTION
**Pull request by terapro**
Branch "tasks 1-2" for assessments and pull request is created.

In this repository the Tasks 1-2 for React are pushed.

It would be very kind of you to leave your comments and useful remarks.

### CODE REVIEW

File: src/index.html [hunk 1]
```diff
@@ -0,0 +1,17 @@
+<!doctype html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <meta name="viewport"
+          content="width=device-width, user-scalable=no, initial-scale=1.0, maximum-scale=1.0, minimum-scale=1.0">
+    <meta http-equiv="X-UA-Compatible" content="ie=edge">
+    <link rel="stylesheet" href="css/bootstrap.min.css" type="text/css">
+    <link rel="stylesheet" href="css/my-styles.css" type="text/css">
```
Comment (alex-belmach) [inline src/index.html]: Styles for components should be imported in components' files, no need to add `link` tags

File: src/css/my-styles.css [hunk 2]
```diff
@@ -0,0 +1,8 @@
+.title {
+  text-align: center;
+  color: blue;
+}
+
+html {
+  margin: 30px;
+}
```
Comment (alex-belmach) [inline src/css/my-styles.css]: Since you're using bootstrap, try to use it's styles unless they're missing in the library

File: src/index.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
+import React, {Component} from 'react';
+import ReactDOM from 'react-dom';
+import "./css/bootstrap.min.css";
+import "./css/my-styles.css";
```
Comment (alex-belmach) [inline src/index.js]: Please use full path unless the file is not in the same folder

File: package.json [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    "mini-css-extract-plugin": "^0.4.0",
+    "react": "^16.4.1",
+    "react-dom": "^16.4.1",
+    "react-router": "^4.3.1",
+    "react-router-dom": "^4.3.1",
+    "redux-devtools": "^3.4.1",
+    "webpack": "^4.12.2",
+    "webpack-cli": "^3.0.8",
+    "webpack-dev-server": "^3.1.4"
+  },
```
Comment (alex-belmach) [inline package.json:32]: I'm suggesting adding something like `eslint` to make your code cleaner

File: src/index.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+const name = "World";
+
+
+// 1. -	React.createElement
```
Comment (alex-belmach) [inline src/index.js]: In real project you commonly should have 1 component in 1 file

File: src/index.js [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+    constructor() {
+        super();
+        this.state = {
+            value: "Click to change my name",
+            buffer: ""
+        };
+        this.setName = this.setName.bind(this);
+        this.setBuffer = this.setBuffer.bind(this);
+    }
+    setBuffer(e){
```
Comment (alex-belmach) [inline src/index.js]: You can define the function as arrow function `setBuffer = () => {}`. So you won't need `.bind` in constructor

File: src/index.js [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+    {className: "title"},
+    "This is a component, created by React.createElement"
+);
+
+// 2. -	functional component
+var el2 =  (name) => (<p>Hello {name}! - functional component</p>);
+
+// 3. -	React.Component
+
+class El3 extends Component {
```
Comment (alex-belmach) [inline src/index.js]: Give components more descriptive name in future

**Review by alex-belmach — COMMENTED** [review]
Please create separate commits for different changes. Use proper naming for the commits. 
For real-project component create folder structure to have a folder for each component and folders for group of components (e.g. for each page).
Also I'm suggesting to start using css preloader from this moment - `sass`, for example.

File: src/index.html (hunk 1 above)
Comment (terapro) [reply src/index.html]: Done

File: src/index.js (hunk 3 above)
Comment (terapro) [reply src/index.js]: Done

File: package.json (hunk 4 above)
Comment (terapro) [reply package.json:32]: The dependancy was added

File: src/index.js (hunk 5 above)
Comment (terapro) [reply src/index.js]: Splited, see the last commits

File: src/index.js (hunk 6 above)
Comment (terapro) [reply src/index.js]: Rewrited

File: src/index.js (hunk 7 above)
Comment (terapro) [reply src/index.js]: Ok

**Pull request merged**